openpype/tools/attribute_defs/widgets.py

```diff
@@ -442,17 +454,40 @@ def _on_value_change(self):
         self.value_changed.emit(new_value, self.attr_def.id)
 
     def current_value(self):
+        if self.multiselection:
+            return self._input_widget.value()
         idx = self._input_widget.currentIndex()
         return self._input_widget.itemData(idx)
 
     def set_value(self, value, multivalue=False):
         if multivalue:
-            set_value = set(value)
-            if len(set_value) == 1:
-                multivalue = False
-                value = tuple(set_value)[0]
+            if self.multiselection:
+                _value = None
+                are_same = True
+                for v in value:
+                    _v = set(v)
+                    if _value is None:
+                        _value = _v
+                    elif not are_same or _value != _v:
+                        _value |= _v
+                        are_same = False
+                if _value is None:
+                    _value = []
+                else:
+                    value = list(_value)
+                multivalue = not are_same
```

Comment:
What godly thing is this doing?

It seems like, if all values are `None` then you want: `[]` (so basically an empty set, but as a list)
If any value is not None then you want the intersections (but as a list)

So you always basically the intersection of all sets combined.
Which is basically this: (assuming each `v` in `value` is an iterable and never e.g. `None`)
```python
value = set.intersection(*list(map(set, value)))
```
Or maybe more readable:
```python
value = set.intersection(*[set(v) for v in value])
```
Or:
```python
sets = [set(v) for v in value]
value = set.intersection(*sets)
```
---
Example:
```python
a = [1, 2, 3]
b = [2, 5]
c = [1, 2, 4]
values = [a, b, c]

def intersection(*iterables):
    sets = [set(it) for it in iterables]
    return set.intersection(*sets)


print(intersection(a, b, c))
# {2}
print(intersection(*values))
# {2}
```

Reply:
How you find out if the values are same using this?

Reply:
How about:
```python
                value_sets = [set(v) for v in value]
                value = list(set.intersection(*value_sets))
                are_same = all(v == value_sets[0] for v in value_sets)
                multivalue = not are_same
```